diffie-hellman-keys.js

```diff
@@ -19,27 +19,28 @@ module.exports = class DHKeys {
   constructor (keys, opts = {}) {
     // feedKeys = { secret: Buffer, public: Buffer }
     this.type = 3 // type: encryption-key
-    this.format = opts.fromEd25519
-      ? 0 // dm related
-      : 1 // probably po-box related
+    this.format = opts.format
+    if (!this.format && opts.fromEd25519) this.format = 0
+    // fromEd25519 implies this is an encryption keypair being derived from a signing keypair
+    // which is the classic way DMs are done (box2-dm-dh format = 0)
 
     if (keys === undefined) return // you are expected to use .generate() in this case
 
-    const secret = keys.secret || keys.private
+    keys.secret = keys.secret || keys.private
```

Comment:
This is a big no. It broke tests elsewhere (where we were doing a deepEquals of the keys object) because now there was a new field attached to the secret object. In general mutation of global state is a big no, and *especially* on a critical piece of info such as `keys`, don't touch `keys` ever, please. Treat it as read-only.